### PR TITLE
Support Deformable Picking and Add Demo for Physics Parameters.

### PR DESCRIPTION
**Pull request by xhan0619**

### CODE REVIEW

File: examples/CommonInterfaces/CommonRigidBodyBase.h [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #include "CommonGUIHelperInterface.h"
 #include "CommonRenderInterface.h"
 #include "CommonCameraInterface.h"
-
+#include "BulletSoftBody/btSoftBody.h"
```
Comment (erwincoumans) [inline examples/CommonInterfaces/CommonRigidBodyBase.h]: Would it be possible to revert this change in CommonRigidBodyBase and move the btSoftBody.h into CommonDeformableBase.h?

File: src/BulletCollision/CollisionDispatch/btCollisionWorld.h [hunk 2]
```diff
@@ -263,6 +263,36 @@ class btCollisionWorld
...
 		}
 	};
+    
+    struct ClosestRayResultCallbackWithInfo : public ClosestRayResultCallback
+    {
+        ClosestRayResultCallbackWithInfo(const btVector3& rayFromWorld, const btVector3& rayToWorld)
+        : ClosestRayResultCallback(rayFromWorld, rayToWorld)
+        {
+        }
+        LocalShapeInfo* m_localShapeInfo;
```
Comment (erwincoumans) [inline src/BulletCollision/CollisionDispatch/btCollisionWorld.h]: this m_localShapeInfo has a very short lifetime, so best not to keep it persistent like this, it is intended only during the callback and not any longer, I think.

Comment (erwincoumans) [reply src/BulletCollision/CollisionDispatch/btCollisionWorld.h]: Also, is is ok to keep this class local to the demo?

**Comment by erwincoumans** [comment]
Some of the demos crashes sometimes when picking (for example cloth friction demo), need to track down why.

File: examples/CommonInterfaces/CommonRigidBodyBase.h (hunk 1 above)
Comment (erwincoumans) [reply examples/CommonInterfaces/CommonRigidBodyBase.h]: The SoftBody demo crashes now (due to changes in CommonRigidBodyBase.h), could you remove those changes (there should be no btSoftBody.h in that class)

**Comment by erwincoumans** [comment]
The crashes are indeed due to storing the pointer to m_localShapeInfo, please store an object instead.
Also, you need to update the pick position to most recent, when setting the constraint.
Remove the changes in CommonRigidBodyBase.

File: src/BulletCollision/CollisionDispatch/btCollisionWorld.h (hunk 2 above)
Comment (xhan0619) [reply src/BulletCollision/CollisionDispatch/btCollisionWorld.h]: Done.

Comment (erwincoumans) [reply src/BulletCollision/CollisionDispatch/btCollisionWorld.h]: the change in btCollisionWorld.h seems to be still there, can you revert that file?

File: examples/CommonInterfaces/CommonRigidBodyBase.h (hunk 1 above)
Comment (erwincoumans) [inline examples/CommonInterfaces/CommonRigidBodyBase.h]: please remove line

File: src/BulletCollision/CollisionDispatch/btCollisionWorld.h (hunk 2 above)
Comment (xhan0619) [reply src/BulletCollision/CollisionDispatch/btCollisionWorld.h]: I already did. I'm fixing another issue revealed by mouse picking. Will push soon.

File: examples/CommonInterfaces/CommonRigidBodyBase.h (hunk 1 above)
Comment (xhan0619) [reply examples/CommonInterfaces/CommonRigidBodyBase.h]: Done.

Comment (xhan0619) [reply examples/CommonInterfaces/CommonRigidBodyBase.h]: Done.

**Comment by erwincoumans** [comment]
Thanks a lot for the CL, LGTM.
